homeassistant/components/lock/zwave.py

```diff
@@ -228,21 +239,29 @@ def update_properties(self):
         _LOGGER.debug("Lock state set from Bool value and is %s", self._state)
         if self.values.access_control:
             notification_data = self.values.access_control.data
-            self._notification = LOCK_NOTIFICATION.get(str(notification_data))
+            self._notification = LOCK_NOTIFICATION.get(
+                str(notification_data))
 
             if self._v2btze:
                 if self.values.v2btze_advanced and \
                         self.values.v2btze_advanced.data == CONFIG_ADVANCED:
-                    self._state = LOCK_STATUS.get(str(notification_data))
+                    self._state = LOCK_STATE_NOTIFICATION.get(
+                        str(notification_data))
                     _LOGGER.debug(
                         "Lock state set from Access Control value and is %s, "
-                        "get=%s", str(notification_data), self.state)
+                        "state=%s", str(notification_data), self._state)
 
         if not self.values.alarm_type:
             return
 
         alarm_type = self.values.alarm_type.data
         _LOGGER.debug("Lock alarm_type is %s", str(alarm_type))
+
+        if alarm_type and alarm_type in LOCK_STATE_ALARM_TYPE:
```

Comment:
Question: We're setting the state based on the alarm type? Does the type change when the lock changes? If that's how it works it could use a comment.